src/cls/IPM/Utils/Module.cls

```diff
@@ -2288,15 +2297,17 @@ ClassMethod %EvaluateSystemExpression(pString As %String) As %String [ Internal
 
 	Set dbRole = ..GetDatabaseRole()
 	Set result = ..%RegExReplace(result, "dbrole", dbRole)
-
-	Set result = ..%RegExReplace(result, "authK5API", $$$AutheK5API)
-	Set result = ..%RegExReplace(result, "authPassword", $$$AutheCache)
-	Set result = ..%RegExReplace(result, "authUnauthenticated", $$$AutheUnauthenticated)
-	Set result = ..%RegExReplace(result, "authLDAP", $$$AutheLDAP)
-	Set result = ..%RegExReplace(result, "authDelegated", $$$AutheDelegated)
-	Set result = ..%RegExReplace(result, "authLoginToken", $$$AutheLoginToken)
-	Set result = ..%RegExReplace(result, "authTwoFactorSMS", $$$AutheTwoFactorSMS)
-	Set result = ..%RegExReplace(result, "authTwoFactorPW", $$$AutheTwoFactorPW)
+	Set result = ..%RegExReplace(result, "globalsDbRole", dbRole)
+
+	// Authentication bits used by Security.Applications
+	Set result = ..%RegExReplace(result, "authK5API", 2**2)
```

Comment:
Open question: for new auth modes, will need to manually add new expressions. Is there a way to not have to update this each time a new auth mode is added?

Reply:
Could code-generate based on contents of %sySecurity.inc but that feels kind of gross.

Reply:
@isc-tleavitt @isc-kiyer 
I'm proposing this:

We can do regex search for patterns like `$$$Authe*` in attributes, and try to evaluate them using the following method. This removes the need to hard-code a list of supported `${auth*}` templates. If the user's system supports the macro, we return its value; otherwise, return 0 (or maybe raise a warning/error)

```objectscript
ClassMethod EvaluateMacro(pMacro As %String) As %String
{
    Set stream = ##class(%Stream.TmpCharacter).%New()
    Do stream.WriteLine("Class Shuheng.Generated Extends %RegisteredObject")
    Do stream.WriteLine("{")
    Do stream.WriteLine("ClassMethod EvalMacro() [ CodeMode = objectgenerator ]")
    Do stream.WriteLine("{")
    Do stream.WriteLine("  Do %code.WriteLine(""#IfDef "_ pMacro _""")")
    Do stream.WriteLine("  Do %code.WriteLine("" Quit $$$"_pMacro_""")")
    Do stream.WriteLine("  Do %code.WriteLine(""#EndIf "")")
    Do stream.WriteLine("  Do %code.WriteLine("" Quit 0 "")")
    Do stream.WriteLine("  Quit $$$OK") 
    Do stream.WriteLine("}")
    Do stream.WriteLine("}")

    Do $system.OBJ.LoadStream(stream, "ck-d")
    Quit $ClassMethod("Shuheng.Generated", "EvalMacro")
    // Maybe delete the helper class after the macro is evaluated
}
```

Reply:
The module.xml will look like
```xml
<WebApplication 
    AutheEnabled="#{$$$AutheCache + $$$AutheUnauthenticated}"
    ... 
/>
```